app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt

```diff
@@ -51,21 +53,34 @@ class PokemonTypeFragment : Fragment() {
 
     private fun showPokemonTypes(pokemonTypes: List<PokemonType>) {
         binding.pokemonAlltypesLoading.visibility = View.GONE
-        binding.pokemonAlltypesRecyclerView.apply {
-            layoutManager =
-                GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
-            adapter = PokemonTypesAdapter(pokemonTypes)
-            visibility = View.VISIBLE
+        pokemonTypes.let {
+            val pokemonTypesAdapter = PokemonTypesAdapter(pokemonTypes, this)
+            binding.pokemonAlltypesRecyclerView.apply {
+                layoutManager =
+                    GridLayoutManager(context, resources.configuration.getColumnsByOrientation(COLUMNS_PORTRAIT, COLUMNS_LANDSCAPE))
+                adapter = pokemonTypesAdapter
+                visibility = View.VISIBLE
+            }
         }
     }
 
     private fun showPokemonTypesError(error: String?) {
         binding.pokemonAlltypesLoading.visibility = View.GONE
-        error?.let { Toast.makeText(context, it, Toast.LENGTH_SHORT).show() }
+        Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
     }
 
     companion object {
```

Comment:
please put the companion at the end of the class

Reply:
done, thanks